### PR TITLE
Some enhancements to the task_retrigger script

### PR DESCRIPTION
**Pull request by colinmacgiolla**
Added some things (listed below) to make the script more general purpose;
- Accepts arguments for username, password and cvp servers
- Handles unverified context for older python versions
- Triggers a compliance check at the beginning (may require a delay?)
- Won't attempt task generation for unregistered devices or devices in the undefined container
- Allows for string match filtering of the devices its going to try and update

### CODE REVIEW

File: docs/labs/lab06-provisioning/vc_task_retrigger.py [hunk 1]
```diff
@@ -4,39 +4,109 @@
...
+        print("Connecting to %s" % cvpserver)
+        try:
+            clnt.connect(nodes=[cvpserver], username=args.username, password=args.password)
+        except Exception as e:
+            print("Unable to connect to CVP: %s" % str(e))
+        #!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+        # Trigger compliance check here
+        #!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!1
+        try:
+            clnt.api.check_compliance('root', 'container')
```
Comment (noredistribution) [inline docs/labs/lab06-provisioning/vc_task_retrigger.py:62]: Maybe we should add a note here? as we don't need to do this post 2020.3.0 where we have continous compliance check, so running `get_device_by_name(device)` or `get_inventory()` will get you always the latest compliance codes

**Review by noredistribution — COMMENTED** [review]
thanks for the PR, LGTM, left a comment

File: docs/labs/lab06-provisioning/vc_task_retrigger.py (hunk 1 above)
Comment (colinmacgiolla) [reply docs/labs/lab06-provisioning/vc_task_retrigger.py:62]: Or just add a check - only do this if the CVP release is <2020.3.0

Comment (noredistribution) [reply docs/labs/lab06-provisioning/vc_task_retrigger.py:62]: good idea! you could use `get_cvp_info()`

Comment (mharista) [reply docs/labs/lab06-provisioning/vc_task_retrigger.py:62]: @colinmacgiolla are you planning to add the version check before I merge this?

**Comment by colinmacgiolla** [comment]
Yes, I'll add it over the weekend

**Review by mharista — CHANGES_REQUESTED** [review]
Github is notifying that the file is missing a newline at the end. Can you add this and then I think we are good to merge.

**Comment by colinmacgiolla** [comment]
Done

**Pull request merged**